user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserOperator.java

```diff
@@ -141,10 +152,95 @@ protected void createOrUpdate(Reconciliation reconciliation, KafkaUser kafkaUser
                 aclOperations.reconcile(KafkaUserModel.getTlsUserName(userName), tlsAcls),
                 aclOperations.reconcile(KafkaUserModel.getScramUserName(userName), scramAcls))
-                .map((Void) null).setHandler(handler);
+                .setHandler(reconciliationResult -> {
+                    StatusUtils.setStatusConditionAndObservedGeneration(kafkaUser, userStatus, reconciliationResult.mapEmpty());
+                    userStatus.setUsername(user.getName());
+
+                    updateStatus(kafkaUser, reconciliation, userStatus).setHandler(statusResult -> {
+                        // If both features succeeded, createOrUpdate succeeded as well
+                        // If one or both of them failed, we prefer the reconciliation failure as the main error
+                        if (reconciliationResult.succeeded() && statusResult.succeeded()) {
+                            createOrUpdateFuture.complete();
+                        } else if (reconciliationResult.failed()) {
+                            createOrUpdateFuture.fail(reconciliationResult.cause());
+                        } else {
+                            createOrUpdateFuture.fail(statusResult.cause());
+                        }
+                        handler.handle(createOrUpdateFuture);
+                    });
+                });
+    }
+
+    protected Future<ReconcileResult<Secret>> reconcileSecretAndSetStatus(String namespace, KafkaUserModel user, Secret desired, KafkaUserStatus userStatus) {
+        return secretOperations.reconcile(namespace, user.getSecretName(), desired).compose(ar -> {
+            Credential credential = null;
+            if (desired != null) {
+                credential = new CredentialBuilder()
+                        .withSecret(desired.getMetadata().getName())
+                        .build();
+            }
+            userStatus.setCredentials(credential);
+            return Future.succeededFuture(ar);
+        });
+    }
+
+    /**
+     * Updates the Status field of the Kafka User CR. It diffs the desired status against the current status and calls
+     * the update only when there is any difference in non-timestamp fields.
+     *
+     * @param kafkaUserAssembly The CR of Kafka user
+     * @param reconciliation Reconciliation information
+     * @param desiredStatus The KafkaUserStatus which should be set
+     *
+     * @return
+     */
+    Future<Void> updateStatus(KafkaUser kafkaUserAssembly, Reconciliation reconciliation, KafkaUserStatus desiredStatus) {
+        Future<Void> updateStatusFuture = Future.future();
+
+        crdOperator.getAsync(kafkaUserAssembly.getMetadata().getNamespace(), kafkaUserAssembly.getMetadata().getName()).setHandler(getRes -> {
+            if (getRes.succeeded()) {
+                KafkaUser user = getRes.result();
+
+                if (user != null) {
+                    if ("kafka.strimzi.io/v1alpha1".equals(user.getApiVersion())) {
```

Comment:
I guess that this kind of test happens for the other status in the other resources as well (Kafka, Kafka Mirror Maker, ...), isn't it better to extract/factor out a util class/method to do this check instead of relying on a hardcoded string in different places?